subsys/bluetooth/mesh/gen_dtt_srv.c

```diff
@@ -188,3 +189,27 @@ int bt_mesh_dtt_srv_pub(struct bt_mesh_dtt_srv *srv,
 
 	return bt_mesh_msg_send(srv->model, ctx, &msg);
 }
+
+struct bt_mesh_dtt_srv *bt_mesh_dtt_srv_get(const struct bt_mesh_elem *elem)
+{
+	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
+	uint16_t index;
+
+	index = elem->addr - comp->elem[0].addr;
+	if (index >= comp->elem_count) {
+		return NULL;
+	}
```

Comment:
This should not be needed. We assume that only correct `elem` is passed to the function.

Reply:
Line 198 is needed or the scene test will fail. And we can not really assume anything as this is a public function call.

Reply:
It is like assuming that NULL pointer will never be passed to the API (but `bt_mesh_model_elem` returns NULL pointer of model wasn't found). It is fine to check invalid parameters if they configure something, but I'm not sure where a user can get invalid elem pointer unless explicitly create one and use it. Right, 198 line can stay.